python/lsst/cp/verify/verifyBfk.py

```diff
@@ -0,0 +1,237 @@
+
+        magnitude = []
+        sizeDiff = []
+        for source, uncorrectedSource, d in matches:
+            # This uses the simple difference in source moments.
+            sourceMagnitude = -2.5 * np.log10(source.getPsfInstFlux())
+            sourceSize = source['base_SdssShape_xx'] + source['base_SdssShape_yy']
+            uncorrectedSize = uncorrectedSource['base_SdssShape_xx'] + uncorrectedSource['base_SdssShape_yy']
+
+            magnitude.append(float(sourceMagnitude))
+            sizeDiff.append(float(uncorrectedSize - sourceSize))
+
+        mask = np.isfinite(magnitude) * np.isfinite(sizeDiff)
+        if 'BRIGHT_SLOPE' in self.config.catalogStatKeywords:
+            exponentialFit = least_squares(modelResidual, [0.0, 0.0, 0.0],
+                                           args=(np.array(magnitude)[mask], np.array(sizeDiff)[mask]),
+                                           loss='cauchy')
+
+            outputStatistics['BRIGHT_SLOPE'] = float(exponentialFit.x[1])
+            outputStatistics['FIT_SUCCESS'] = exponentialFit.success
+            self.debugFit('brightSlope', magnitude, sizeDiff, exponentialFit.x)
+
+        outputStatistics['MAGNITUDES'] = magnitude
+        outputStatistics['SIZE_DIFF'] = sizeDiff
+
+        # This should have a debug view
+        return outputStatistics
+
+    def debugFit(self, stepname, xVector, yVector, yModel):
+        """Debug method for linearity fitting.
+        Parameters
+        ----------
+        stepname : `str`
+            A label to use to check if we care to debug at a given
+            line of code.
+        xVector : `numpy.array`, (N,)
+            The values to use as the independent variable in the
+            linearity fit.
+        yVector : `numpy.array`, (N,)
+            The values to use as the dependent variable in the
+            linearity fit.
+        yModel : `numpy.array`, (N,)
+            The values to use as the linearized result.
+        """
+        frame = getDebugFrame(self._display, stepname)
+        if frame:
+            import matplotlib.pyplot as plt
+            figure, axes = plt.subplots(1)
+
+            axes.scatter(xVector, yVector, c='blue', marker='+')
+            modelX = np.arange(np.min(xVector) - 1, np.max(xVector) + 1, 0.1)
+            axes.plot(modelX, exponentialModel(modelX, *yModel), 'r-')
+            plt.xlabel("Instrumental PSF magnitude")
+            plt.ylabel("Source size trace")
+            plt.title(f"BFK slope: {yModel[0]:.3f} + {yModel[1]:.3f} m")
+            figure.show()
+            prompt = "Press Enter or c to continue [chp]..."
+
+            while True:
+                ans = input(prompt).lower()
+                if ans in ("", " ", "c",):
+                    break
+                elif ans in ("p", ):
+                    import pdb
+                    pdb.set_trace()
+                elif ans in ("h", ):
+                    print("[h]elp [c]ontinue [p]db e[x]itDebug")
```

Comment:
I'm still sad we haven't centralised this (and added a `try: import ipdb as pdb; except import pdb` block in the process).